### PR TITLE
Adding else remove comments

### PR DESCRIPTION
**Pull request by colleend**
This PR:

* un-comments portions of `annotated_lambda` and adds in conflict resolution
* adds the case where else comes after a newline following the `if` control body into scanner.cc

Increases parsing rate to 98.980% after Pad's PR!

To test, run `npx tree-sitter test`

cc: @aryx cc: @fwcd

### CODE REVIEW

**Comment by aryx** [comment]
woohoo, we just miss 0.020% to get to 99%!

**Comment by aryx** [comment]
@fwcd can we get this merged? We have a deadline for reaching 99% by next week.

File: src/scanner.c [hunk 1]
```diff
@@ -51,6 +51,14 @@ static bool scan_whitespace_and_comments(TSLexer *lexer) {
   }
 }
 
+bool scan_for_word(TSLexer *lexer, char* word, unsigned len) {
+    advance(lexer);
+    for (unsigned i = 0; i < len; i++) {
```
Comment (aryx) [inline src/scanner.c:56]: hmm what happens if the file contains less than len characters remaining? What does
advance(lexer) do when there is no more characters to read?

**Review by aryx — CHANGES_REQUESTED** [review]

File: src/scanner.c (hunk 1 above)
Comment (colleend) [reply src/scanner.c:56]: Good question -> the first `advance(lexer)` in the function advances past the `e` character (or another character chosen) in the switch statement (so not a problem), and the `lexer->lookahead` check in the if statement catches whether the file ends after.

I tested this on a file with just the `e` and an empty file, and both of them parse correctly 👍

Comment (aryx) [reply src/scanner.c:56]: perfect.

**Review by aryx — APPROVED** [review]

**Comment by colleend** [comment]
I also have some more small fixes that get us to 99.04%! After this gets merged, I'll make that PR :D

**Pull request merged**